src/components/CutoutImage/CutoutImage.tsx

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import {
+  CutoutBottomLeft,
+  CutoutBottomRight,
+  CutoutHighBottomLeft,
+  CutoutHighBottomRight,
+  CutoutHighTopLeft,
+  CutoutHighTopRight,
+  CutoutTopLeft,
+  CutoutTopRight,
+  CutoutWideBottomLeft,
+  CutoutWideBottomRight,
+  CutoutWideTopLeft,
+  CutoutWideTopRight,
+  ImageContainer,
+} from './styled';
+
+type CutoutImageProps = {
+  mode: string;
+  image: string;
+};
+
+function CutoutImage({ mode, image }: CutoutImageProps) {
+  switch (mode) {
+    case 'TopLeft':
+      return (
+        <ImageContainer>
+          <CutoutTopLeft src={image} />
+        </ImageContainer>
+      );
+    case 'TopRight':
+      return (
+        <ImageContainer>
+          <CutoutTopRight src={image} />
+        </ImageContainer>
+      );
+    case 'BottomLeft':
+      return (
+        <ImageContainer>
+          <CutoutBottomLeft src={image} />
+        </ImageContainer>
+      );
+    case 'BottomRight':
+      return (
+        <ImageContainer>
+          <CutoutBottomRight src={image} />
+        </ImageContainer>
+      );
+    case 'WideTopLeft':
+      return (
+        <ImageContainer>
+          <CutoutWideTopLeft src={image} />
+        </ImageContainer>
+      );
+    case 'WideTopRight':
+      return (
+        <ImageContainer>
+          <CutoutWideTopRight src={image} />
+        </ImageContainer>
+      );
+    case 'WideBottomLeft':
+      return (
+        <ImageContainer>
+          <CutoutWideBottomLeft src={image} />
+        </ImageContainer>
+      );
+    case 'WideBottomRight':
+      return (
+        <ImageContainer>
+          <CutoutWideBottomRight src={image} />
+        </ImageContainer>
+      );
+    case 'HighTopLeft':
+      return (
+        <ImageContainer>
+          <CutoutHighTopLeft src={image} />
+        </ImageContainer>
+      );
+    case 'HighTopRight':
+      return (
+        <ImageContainer>
+          <CutoutHighTopRight src={image} />
+        </ImageContainer>
+      );
+    case 'HighBottomLeft':
+      return (
+        <ImageContainer>
+          <CutoutHighBottomLeft src={image} />
+        </ImageContainer>
+      );
+    case 'HighBottomRight':
+      return (
+        <ImageContainer>
+          <CutoutHighBottomRight src={image} />
+        </ImageContainer>
+      );
+    default:
+      return <h3>Failed to place cutoff image</h3>;
+  }
```

Comment:
Can't think of any better solution on the spot, but it would be nice if we could avoid having large switch-statement like this 🤔.

Reply:
The main purpose here is to make it generic, and to have 12 (or more for that matter 😇) defined types of CutoutImages.

I agree that the switch-statement is a bit large. The switch statement became large because I though it was better to make the Images inherit from ImageContainer, so that you only have to change 2 variables if you want to change the image size.

One quick solution here is to hardcode the 4 CutoutImages (if we know we only need 4), but obviously that isnt so scalable😅

Reply:
No worries, I understand 😆. Was just wondering because I've seen this issue before, but never had time to think hard about if a prettier solution exists 😔